### PR TITLE
Add new About Dynamic Consensus topic; clean up terms

### PR DESCRIPTION
**Pull request by chenette**
Add new topic "About Dynamic Consensus"
- Summarizes admin tasks for setting consensus in a new network
   and for changing consensus on a running network.
- For each consensus type, lists requirements and on-chain settings
- Adds links to this topic from the Introduction, App Dev Guide (Ubuntu network
   procedure) and Sys Admin Guide
- Documents consensus backwards compatibility - includes information about
   consensus defaults                                                            
 
Clean up consensus terms in Sawtooth docs
- Change "dev mode" to "Devmode" to match the consensus engine name
- Make other terms consistent and coherent (in the glossary and elsewhere)

### CODE REVIEW

File: docs/source/architecture/journal.rst [hunk 1]
```diff
@@ -310,8 +310,10 @@ necessary transaction processors must be running.
 If any transactions in the genesis block set or change settings, Sawtooth
 requires the `Sawtooth Settings transaction processor <../cli/settings-tp>`_
 or an equivalent implementation.
-For example, if the genesis block configures PoET consensus, this transaction
-processor is handles the transactions with PoET settings.
+For example, the genesis block specifies the :term:`consensus engine` and
+related settings (unless :term:`Devmode consensus` is used), so the Settings
```
Comment (ltseeley) [inline docs/source/architecture/journal.rst]: Why "unless Devmode consensus is used"?

File: docs/source/sysadmin_guide/about_dynamic_consensus.rst [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+   Sawtooth provides defaults for backward compatibility with previous versions
+   of Sawtooth (1.0 and earlier), when Devmode and PoET were the only available
+   consensus algorithms.
+
+   * If ``sawtooth.consensus.algorithm.version`` is not set, the default version
+     is ``0.1``.
+
+   * If ``sawtooth.consensus.algorithm.name`` is not set, Sawtooth checks for
+     the deprecated setting ``sawtooth.consensus.algorithm``, which was used in
+     previous releases to specify PoET consensus. If this setting exists,
```
Comment (ltseeley) [inline docs/source/sysadmin_guide/about_dynamic_consensus.rst]: This isn't 100% accurate; if this setting exists, it'll use that as the name. If this setting isn't `poet`, Sawtooth won't use PoET.

File: docs/source/sysadmin_guide/about_dynamic_consensus.rst [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  (such as PoET Validator Registry).
+
+The rest of this topic summarizes the settings and requirements for each
+available consensus type: :ref:`PBFT <pbft-consensus-label>`,
+:ref:`PoET <poet-consensus-label>`, :ref:`Raft <raft-consensus-label>`, and
+:ref:`Devmode <devmode-consensus-label>`.
+
+.. note::
+
+   **Default Consensus Settings:**
```
Comment (ltseeley) [inline docs/source/sysadmin_guide/about_dynamic_consensus.rst]: I think it would be worth adding something to the beginning of this section that the `sawtooth.consensus.algorithm.name/version` settings are required. for current versions; if you only read this, you might think what's explained below is something you could use, but you can't. Also, "Default Consensus Settings" probably isn't the best term to use, because there's not "default" consensus (as you explain below). "Backwards Compatibility of Consensus Settings" or something similar would be better.

File: docs/source/sysadmin_guide/about_dynamic_consensus.rst [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  See :doc:`/sysadmin_guide/creating_genesis_block` for more information on the
+  version and validator public keys for the PBFT member list .
+
+
+.. _poet-consensus-label:
+
+PoET Consensus
+==============
+
+:term:`PoET <PoET consensus>` is a Nakamoto-style (voting) consensus algorithm
```
Comment (ltseeley) [inline docs/source/sysadmin_guide/about_dynamic_consensus.rst]: Should be "lottery", not "voting"

File: docs/source/sysadmin_guide/about_dynamic_consensus.rst [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+* Use these on-chain settings for the Devmode consensus engine:
+
+  .. code-block:: none
+
+     sawtooth.consensus.algorithm.name=Devmode
+     sawtooth.consensus.algorithm.version=0.1
+
+  .. note::
+
+     These settings are optional, because Devmode version 0.1 is the default
```
Comment (ltseeley) [inline docs/source/sysadmin_guide/about_dynamic_consensus.rst]: These settings are **not** optional; the genesis validator will fail if they are not specified in 1.2. The backwards compatibility "defaults" are only for the purpose of backwards compatibility. All new networks must explicitly have the name and version settings.

File: docs/source/architecture/journal.rst (hunk 1 above)
Comment (danintel) [reply docs/source/architecture/journal.rst]: I wondered if there's any DevMode-specific settings, and it seems there are:
```
sawtooth.consensus.max_wait_time
    Maximum devmode consensus wait time, in seconds
sawtooth.consensus.min_wait_time
    Minimum devmode consensus wait time, in seconds
```
Unfotunately, they don't have `devmode` in the name to make it obvious they are DevMode-specific

Comment (chenette) [reply docs/source/architecture/journal.rst]: Thanks, @danintel - I'll add these settings to the Devmode section.

File: docs/source/sysadmin_guide/about_dynamic_consensus.rst [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+  transaction processors required for consensus (such as PoET Validator
+  Registry).
+  See :doc:`/sysadmin_guide/setting_up_sawtooth_poet-sim` for this procedure.
+
+* To change consensus for a running network, administrators must coordinate the
+  changes for all nodes. Each administrator must install and start the new
+  consensus engine. Then one administrator submits a transaction to change the
+  on-chain consensus settings.
+
+  After the consensus has changed, each administrator can stop the previous
```
Comment (agunde406) [inline docs/source/sysadmin_guide/about_dynamic_consensus.rst]: It is worth mentioning that if a node is added or is catching up, they would need to have the correct consensus engines/tp connected to handle all the previous versions of consensus used to build the chain. For example if the first half of the chain used  x and then switched to y a new node would need x and y until it caught up to the other nodes.

File: docs/source/sysadmin_guide/about_dynamic_consensus.rst [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+     sawtooth.consensus.algorithm.name=PoET
+     sawtooth.consensus.algorithm.version=0.1
+     sawtooth.poet.report_public_key_pem="$(cat * /etc/sawtooth/simulator_rk_pub.pem)"
+     sawtooth.poet.valid_enclave_measurements=$(poet enclave measurement)
+     sawtooth.poet.valid_enclave_basenames=$(poet enclave basename)
+
+  .. note::
+
+     For a PoET test network with less than 10 or 12 nodes, disable the
+     defense-in-depth tests (ztest) with these settings:
```
Comment (agunde406) [inline docs/source/sysadmin_guide/about_dynamic_consensus.rst]: These settings disable the z test, c test and k test not just the ztest.

File: docs/source/architecture/journal.rst (hunk 1 above)
Comment (chenette) [reply docs/source/architecture/journal.rst]: @danintel - Your wonderful FAQ also lists this setting for Devmode:

`sawtooth.consensus.valid_block_publishers`
    `List of public keys for allowed block publishers. For devmode`

@ltseeley - Do you know if this setting is used in Sawtooth 1.2?

Comment (ltseeley) [reply docs/source/architecture/journal.rst]: It's not used in 1.1 or 1.2; the validator code it's used in is actually dead code. It's not in the Devmode engine either, so it looks like that's deprecated.

File: docs/source/sysadmin_guide/about_dynamic_consensus.rst (hunk 6 above)
Comment (chenette) [reply docs/source/sysadmin_guide/about_dynamic_consensus.rst]: 💡

Comment (chenette) [reply docs/source/sysadmin_guide/about_dynamic_consensus.rst]: Would it make sense to tell admins to just keep the old engines/TPs running in case a node is added later?  (And, of course for slow nodes who aren't caught up yet.)

Comment (ltseeley) [reply docs/source/sysadmin_guide/about_dynamic_consensus.rst]: Well technically, the node that's caching up would only need x until it reaches the block where the network switched to y. To answer your question, @chenette, existing nodes won't have to keep old stuff running; when a new node is added, it will have to run the old engine + TPs.

**Comment by chenette** [comment]
I pushed some fixup commits with the requested changes, plus a rewrite for the intro of "About Dynamic Consensus".  

- This topic now starts with three bullet points that summarize the consensus engine, consensus endpoint, and required consensus settings. 

- I moved the bullets on setting/changing consensus into their own subsections, "Configuring consensus for a New Network" and "Changing Consensus on a Running Network". I also expanded each paragraph into a summary of the steps (now numbered) with additional notes, tips, and links to the actual procedures.

File: docs/source/sysadmin_guide/about_dynamic_consensus.rst [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+     is ``0.1``.
+
+   * If ``sawtooth.consensus.algorithm.name`` is not set, Sawtooth checks for
+     the deprecated setting ``sawtooth.consensus.algorithm``. If this setting
+     exists, Sawtooth uses the specified consensus.
+
+   * Otherwise, Sawtooth uses Devmode consensus.
+
+   **Important:** A consensus engine is always required. (There is no default
+   consensus engine.)
```
Comment (danintel) [inline docs/source/sysadmin_guide/about_dynamic_consensus.rst]: This is ambiguous to me. There is a default consensus algorithm (Devmode). But a separate consensus engine process must always be ran--and I think this statement is referring to the engine process.  Perhaps change this to something like:

> **Important:** A consensus engine is always required _to be running on all nodes_. (There is no default consensus engine.)

Comment (chenette) [reply docs/source/sysadmin_guide/about_dynamic_consensus.rst]: @danintel - How about this wording?

*Important:* In release 1.1 and later, a consensus engine is always required. The same consensus engine must be running on all nodes in the network.

Comment (danintel) [reply docs/source/sysadmin_guide/about_dynamic_consensus.rst]: Looks good. Thanks!
